src/Telemetry/Library/OutOfProcess/SimpleYamlSerializer.cs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Applications.Events;
+
+namespace Microsoft.Quantum.Telemetry.OutOfProcess
+{
+    internal class SimpleYamlSerializer : IOutOfProcessSerializer
+    {
+        private static readonly string LineBreak = @"__\r\n__";
+        private static readonly string EventNamePropertyName = "__name__";
+        private static readonly Regex CommandRegex = new(@"^((?<command>- command:\s+!(?<commandType>[^\s$]+).*$)|(?<property>\s{4}(?<key>[^\s:]+):\s*(!(?<type>[^\s:\+]+)(?<pii>\+Pii)?\s+)?(?<value>.+)))$", RegexOptions.Compiled);
```

Comment:
A compiled Regex should perform as good as if we were to build our own state machine parser.
The slow down could come from looking for the capture groups by name in the Match object, but since this is running out of process anyway, I didn't want to over optimize it.